index.bs

```diff
@@ -11302,6 +11306,153 @@ The [=remote end steps=] given |session| and |command parameters| are:
+
+<pre class="cddl remote-cddl">
+ExtensionsCommand = (
+  extensions.Install
+)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+ExtensionsResult = {
+  extensions.InstallResult
+}
+</pre>
+
+### Types ### {#module-extensions-types}
+
+#### The extensions.Extension Type #### {#type-extensions-Extension}
+
+<pre class="cddl local-cddl">
+extensions.ExtensionArchive = {
+   files: [+text],
+}
+</pre>
+
+<pre class="cddl remote-cddl local-cddl">
+extensions.Extension = {
+   installPath: extensions.ExtensionPath / extensions.ExtensionArchivePath / extensions.ExtensionBase64Encoded,
+   ? allowPrivateBrowsing: bool .default false,
+}
+
+extensions.ExtensionPath = {
+   type: "path",
+   path: text,
+}
+
+extensions.ExtensionArchivePath = {
+   type: "archive-path",
+   path: text,
+}
+
+extensions.ExtensionBase64Encoded = {
+   type: "base64",
+   value: text,
+}
+</pre>
+
+The <code>extensions.Extension</code> type represents an extension with its additional properties.
+
+<div algorithm>
+To <dfn>expand a install path spec</dfn> given |install path spec|:
+
+1. Switch on the value of the <code>type</code> field of |install path spec|:
+
+   <dl>
+    <dt>"<code>path</code>"
+    <dd>Let |path| be a value of <code>path</code> field of |install path spec|.
+
+    <dt>"<code>archive-path</code>"
+    <dd>
+      1. Let |archive path| be a value of <code>path</code> field of |install path spec|.
+      1. Perform any implementation-defined steps to extract the archive from <code>archive path</code> to a temporary directory and let |path| be the path to that temporary directory.
+
+    <dt>"<code>base64</code>"
+    <dd>
+      1. Let |archive path| be a value of <code>path</code> field of |install path spec|.
+      1. Perform any implementation-defined steps to decode <code>archive path</code> to a zip archive and extract that archive to a temporary directory and let |path| be the path to that temporary directory.
+
+   </dl>
+
+1. Let |extension files| be a result of implementation-defined steps to list all paths in |path| folder.
+
+1. Let |extension archive| be a map matching the <code>extensions.ExtensionArchive</code> production, with the <code>files</code> field set to |extension files|.
+
+1. Return [=success=] with data |extension archive|.
+
+</div>
+
+### Commands ### {#module-extensions-commands}
+
+#### The extensions.install Command #### {#command-extensions-install}
+
+The <dfn export for=commands>extensions.install</dfn> command installs an extension for the session.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+     <pre class="cddl remote-cddl">
+     extensions.Install = (
+       method: "extensions.install",
+       params: extensions.InstallParameters
+     )
+
+     extensions.InstallParameters = {
+       extension: extensions.Extension,
+     }
+     </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+      extensions.InstallResult = (
+        extensionId: text
```

Comment:
We don't use the `fooId` pattern in general; wherever possible we just use `foo`, so this should just be `extension`.

Reply:
I've renamed it to the `extensions` but must say that `extension id` is something that will be well understood by the extension developers. Each browser extension has own ID:
 
<img width="379" alt="Screenshot 2024-10-11 at 16 39 46" src="https://github.com/user-attachments/assets/df5803df-a2fa-4dca-9d37-ab1ff76fab6d">

and if Firefox the ID can be specified in the manifest.json

```json
"browser_specific_settings": {
  "gecko": {
    "id": "firefox@ghostery.com",
    "strict_min_version": "112.0"
  }
}
```

How about we rename it from `extension` to `id`?

Reply:
`id` would be much worse :)

The point is that within bidi we don't generally use the `Id` suffix when passing back a handle to something. We just name the thing e.g. `context` rather than `contextId`, `request` rather than `requestId`, etc. There are three exceptions I can see: script objects get both `sharedId` and `internalId`, and the session itself gets `sessionId` for compatibility with WebDriver classic. So by the same token a handle identifying a specific web extension should just be called `extension` in the protocol.

That said I don't really feel strongly, so if others think this is a case where it's worth breaking the convention, I won't object.

Reply:
I, for one, would prefer to continue with convention, if only for the sake of consistency.